---
title: "Admixture analysis"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## ngsAdmix

Use the [run_ngsadmix.sh](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/scripts/run_ngsadmix.sh) script to run an admixture analysis using ngsAdmix. A range of files and parameters have to be provided in the following order:

+ The project's base directory (`BASEDIR`), e.g. `path/base_directory/`
+ Path to beagle formatted genotype likelihood file (`BEAGLE`), e.g. `path/genotype_likelihood.beagle.gz`
+ Minimum minor allele frequency (`MINMAF`), e.g. `0.05`
+ Minimum number of K (`MINK`), e.g. `1`
+ Maximum number of K (`MAXK`), e.g. `10`

Run the script using the following command with nohup from the script directory:

``` {bash eval=FALSE}
nohup ./run_ngsadmix.sh \
BASEDIR \
BEAGLE \
MINMAF \
MINK \
MAXK \
> path/output_logfile.nohup &
```

## PCAngsd

Use the [run_pcangsd.sh](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/scripts/run_pcangsd.sh) to run an admixture analysis with PCAngsd. A range of files and parameters have to be provided in the following order:

+ The project's base directory (`BASEDIR`), e.g. `path/base_directory/`
+ Path to beagle formatted genotype likelihood file (`BEAGLE`), e.g. `path/genotype_likelihood.beagle.gz`
+ Minor allele frequency filter (`MINMAF`), e.g. `0.05`
+ Type of analysis to run: for admixture use `admix` (other options: pca, selection, inbreedSites, kinship)
+ Minimum number of eigenvectors to use in the modelling of individual allele frequencies (`MINE`), e.g. `1`
+ Maximum number of eigenvectors to use in the modelling of individual allele frequencies (`MAXE`), e.g. `10`

Run the script using the following command with nohup from the script directory:

``` {bash eval=FALSE}
nohup ./run_pcangsd.sh \
BASEDIR \
BEAGLE \
MINMAF \
admix \
MINE \
MAXE > path/output_logfile.nohup &
```
