---
title: "Minor allele frequency estimation"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Instructions

Use the [get_maf_per_pop.sh](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/scripts/get_maf_per_pop.sh) script to get minor allele frequency (MAF) estimates for distinct sites (e.g. sites file from SNP calling script) for individual groups of individuals, i.e. populations. This script loops over populations as provided in the sample table. The following additional (not explained above) parameters and files have to be provided:

+ The project's base directory (`BASEDIR`), e.g. `path/base_directory/`
+ Path to a tab deliminated sample table (`SAMPLETABLE`) where the 4th column is the sample ID, the 2nd column is the lane number, and the 3rd column is sequence ID. The combination of these three columns have to be unique, which then forms the 1st column in the format of sampleID_seqID_laneID. The 6th column should be data type, which is either pe or se. This is the same as the merged sample table used in [data-processing](https://github.com/therkildsen-lab/greenland-cod/blob/master/markdowns/data_processing.md). e.g. `path/sample_table.tsv`
+ Index of the column with population information in the sample table (`POPCOLUMN`), e.g. `5`
+ Prefix in file name of the bamfile list (`BAMLISTPREFIX`), e.g. `bam_list_realigned_`
+ Path to the indexed reference genome (`REFERENCE`), e.g. `path/reference_genome.fasta`
+ Path to the SNP list (`SNPLIST`), e.g. `path/global_snp_list.txt`
+ Minimum combined sequencing depth in a population (`MINDP`)
+ Maximum combined sequencing depth across all individual in a population (`MAXDP`)
+ Minimum number of individuals a read has to be present in a population (`MININD`)
+ The minimum base quality score (`MINQ`), e.g `20`

Run the script using the following command with nohup from the script directory:

``` {bash eval=FALSE}
nohup ./get_maf_per_pop.sh \
BASEDIR \
SAMPLETABLE \
POPCOLUMN \
BAMLISTPREFIX \
REFERENCE \
SNPLIST \
MINDP \
MAXDP \
MININD \
MINQ \
> path/output_logfile.nohup &
```

## Note

Important is that one uses `-doMajorMinor 3` when providing a sites file to use the provided major and minor allele as the basis for estimating minor allele frequencies. 
