---
title: "Population genomic analysis of low-coverage whole genome data"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Overview

This collection of scripts cover the data analysis part of the low-coverage whole genome sequencing pipeline (i.e. bottom part of the following diagram), going from quality checked bam files to various population genomic inference. It is primarily ANGSD-based, but will also incorporate other downstream analysis software programs that can take ANGSD files at input. 

We aim to generate a pipeline that is user-friendly and highly flexible (e.g. it should be compatible to most study organisms, sequencing configuration, and computing server structure). However, this is still a work in progress. Please submit a GitHub issue if you encounter any problems, and we will try to get back to you in a timely fashion. 

```{r echo=FALSE}
knitr::include_graphics("misc/fig2_lcwgs_guide.png")
```


## Before you start

We recommend you to have a project base directory that contains the following subdirectories:

  * `bam` bam files
  
  * `nohups` nohup log files
  
  * `sample_lists` small text files, such as lists of bam files
   
  * `markdowns` markdown files tracking the workflow; it is recommended that you first create Rmd files using Rstudio, and knit these into GitHub markdown format

  * `angsd` angsd output files
  
  * `scripts` scripts specific to your projects (e.g. merging certain bam files)

