---
title: "localPCA"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Instructions

First, use the [subset_beagle_by_lg.sh](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/scripts/subset_beagle_by_lg.sh) script to subset the beagle files by LGs or chromosomes. A range of files and parameters have to be provided in the following order:
+ Path to a genome-wide beagle file (`BEAGLE`), e.g. `/workdir/cod/greenland-cod/angsd/bam_list_realigned_mindp161_maxdp768_minind97_minq20.beagle.gz`
+ Path to a list of LGs or chromosomes that you want to subset by (`LGLIST`), e.g. `/workdir/cod/greenland-cod/sample_lists/lg_list.txt`

Run the script using the following command with nohup from the script directory:

```{bash eval=FALSE}
nohup ./subset_beagle_by_lg.sh \
BEAGLE \
LGLIST \
> path/output_logfile.nohup &
```
Then, use the [run_local_pca.sh](https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/scripts/run_local_pca.sh) script to run a localPCA analysis. A range of files and parameters have to be provided in the following order:
+ Path to a beagle.gz file that you have used for the subsetting step (`BEAGLE`), e.g. `/workdir/cod/greenland-cod/angsd/bam_list_realigned_mindp161_maxdp768_minind97_minq20.beagle.gz`
+ Path to a list of LGs or chromosomes that you have used for the subsetting step (`LGLIST`), e.g. `/workdir/cod/greenland-cod/sample_lists/lg_list.txt`
+ Number of SNPs to include in each window (`SNP`), e.g. 10000
+ Number of PCs to keep for each window (`PC`), e.g. 2

Run the script using the following command with nohup from the script directory:

```{bash eval=FALSE}
nohup ./run_local_pca.sh \
BEAGLE \
LGLIST \
SNP \
PC \
> path/output_logfile.nohup &
```