<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>EEMS</title>

<script src="site_libs/header-attrs-2.9/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Data analysis pipelines with genomic data</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    lcWGS pipeline
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Pipeline for low-coverage whole genome sequencing data</li>
    <li>
      <a href="lcwgs_data_analysis.html">Overview</a>
    </li>
    <li>
      <a href="angsd_global_snp_calling.html">SNP calling</a>
    </li>
    <li>
      <a href="ld.html">Linkage disequilibrium</a>
    </li>
    <li>
      <a href="pca.html">Individual-level PCA and PCoA</a>
    </li>
    <li>
      <a href="local_pca.html">localPCA</a>
    </li>
    <li>
      <a href="admixture.html">Admixture analysis</a>
    </li>
    <li>
      <a href="relatedness.html">Relatedness &amp; pedigree</a>
    </li>
    <li>
      <a href="selection_scan.html">Selection scan</a>
    </li>
    <li>
      <a href="gwas.html">GWAS</a>
    </li>
    <li>
      <a href="diversity_per_ind.html">Individual-level genetic diversity</a>
    </li>
    <li>
      <a href="get_maf_per_pop.html">Minor allele frequency estimation</a>
    </li>
    <li>
      <a href="fst.html">Fst estimation</a>
    </li>
    <li>
      <a href="dxy.html">dxy estimation</a>
    </li>
    <li>
      <a href="eems.html">EEMS</a>
    </li>
    <li>
      <a href="diversity_per_pop.html">Population-level genetic diversity and neutrality stats</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">EEMS</h1>

</div>


<div id="instructions" class="section level2">
<h2>Instructions</h2>
<p>Estimating Effective Migration Surfaces. These scripts are for the use of a modifief version that works with a pairwise Fst matrix rather than individual pairwise genetic dissimilarity estimates. See <a href="https://github.com/dipetkov/eems">EEMS github page</a> for details.</p>
<p>Required input files:</p>
<ol style="list-style-type: decimal">
<li>datapath.diffs: Symmetric Fst matrix with zeros on the diagonal.</li>
<li>datapath.coord: the sample coordinates (longitude and latitude coordinates per sample (tab separated), one sample per line)</li>
<li>datapath.outer: the habitat coordinates (as a sequence of vertices that outline a closed polygon)</li>
</ol>
<p>Habitat outline can be created manually as a “polyline” on this <a href="http://www.birdtheme.org/useful/v3tool.html">site</a>. The habitat vertices should be listed counterclockwise and form a closed polygon.</p>
<p>Furthermore, a parameter file (e.g. params-simno1.ini) has to be created that provides paths to the input files and additional parameters. The file should look like this:</p>
<pre><code>datapath = ./workdir/arne/results/mme_popstructure/eems_res/silverside_eems
mcmcpath = /workdir/arne/results/mme_popstructure/eems_res/silverside_eems
nIndiv = 11
nSites = 4696248 
nDemes = 1000
diploid = true
numMCMCIter = 2000000
numBurnIter = 1000000
numThinIter = 9999</code></pre>
<p>This file specifies the required arguments: the path to the input data (datapath) the path to the output data (mcmcpath) the number of samples (nIndiv) the number of markers (nSites) the density of the population grid (nDemes) is the species diploid or haploid (diploid) the number of MCMC and burn-in iterations (numMCMCIter, numBurnIter) and the thinning interval (numThinIter).</p>
<p>EEMS can be run using the <a href="https://github.com/therkildsen-lab/genomic-data-analysis/blob/master/scripts/run_eems_fst.sh">run_eems_fst.sh</a> script with the specified parameter file</p>
<pre class="bash"><code>nohup sh run_eems_fst.sh \
/path/parameterfile.ini \
&gt; /path/logfiles/nohup.log &amp;</code></pre>
<p>The EEMS results can be plotted using the rEEMSplots R-package.</p>
<p>Commands for installing the rEEMSplots R-package and dependencies:</p>
<pre class="r"><code>install.packages(c(&quot;Rcpp&quot;,&quot;RcppEigen&quot;,&quot;raster&quot;,&quot;rgeos&quot;,&quot;sp&quot;,&quot;rgdal&quot;,&quot;rworldmap&quot;,&quot;rworldxra&quot;))

if (file.exists(&quot;./rEEMSplots/&quot;)) {
  install.packages(&quot;rEEMSplots&quot;, repos = NULL, type = &quot;source&quot;)
} else {
  stop(&quot;Move to the directory that contains the rEEMSplots source to install the package.&quot;)
}</code></pre>
<p>There are many different options for modifying the resulting plots. The following code shows how to plot the results including a map of the study area, grid and migration rate estimates. The package also automatically plots other diagnostic plots. More details can be found on the EEMS github page.</p>
<pre class="r"><code>library(rEEMSplots)
library(&quot;rgdal&quot;)
library(&quot;rworldmap&quot;)
library(&quot;rworldxra&quot;)

mcmcpath = &quot;/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/mme_popstructure/eems/silverside_eems&quot;
plotpath = &quot;/Users/arnejacobs/Dropbox/Cornell_Postdoc/analysis/results/mme_popstructure/eems/silverside_eems&quot;

projection_none &lt;- &quot;+proj=longlat +datum=WGS84&quot;
projection_mercator &lt;- &quot;+proj=merc +datum=WGS84&quot;

eems.plots(mcmcpath, plotpath, longlat = TRUE
           ,out.png=FALSE, add.grid = TRUE
           ,col.grid = &quot;gray90&quot;,
           ,add.demes = TRUE, remove.singletons = FALSE
           ,projection.in = projection_none
           ,projection.out = projection_mercator
           ,add.map = TRUE, 
           ,col.map = &quot;black&quot;
           ,lwd.map = 1)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
